frontend/src/component/LinkForm.tsx

```diff
@@ -0,0 +1,37 @@
+import {FormEvent, useState} from "react";
+import axios from "axios";
+
+export default function LinkForm() {
+
+    const [file, setFile] = useState<File | undefined>(undefined);
+
+    const handleFileUpload = async (event: FormEvent<HTMLFormElement>) => {
+        event.preventDefault();
+        if (!file) {
+            alert('Bitte wähle eine Datei aus.');
+            return;
+        }
+        const formData = new FormData();
+        formData.append('file', file);
+
+        axios.post("/api/upload", formData, {
+            headers: { 'Content-Type': 'multipart/form-data' },
+        })
+            .then(response => console.log(response))
+            .catch(error => {console.error(error)});
+    }
+
+
+    return (
+        <>
+            <form onSubmit={handleFileUpload}>
+                <input
+                    type={"file"}
+                    accept={"audio/mp3, audio/mp4, audio/m4a"}
+                    onChange={(event) => setFile(event.target.files?.[0])}
```

Comment:
If the user deselects the file (or cancels the file selection dialog), `event.target.files` is `null`, meaning:

- `event.target.files?.[0]` evaluates to `undefined`.
- `setFile(undefined)` is never explicitly called.
- The state remains stuck on the previous file instead of resetting.

```suggestion
                    onChange={(event) => setFile(event.target.files?.[0] || undefined)}
```